### PR TITLE
feat: add ext packages

### PR DESCRIPTION
**Pull request by eddycharly**
Add ext packages.

### CODE REVIEW

File: ext/resource/convert/convert_test.go
Comment (realshuting) [inline ext/resource/convert/convert_test.go:1]: Is this no longer valid?

File: ext/resource/loader/loader_test.go
Comment (realshuting) [inline ext/resource/loader/loader_test.go:1]: Scheduled to be removed?

**Review by realshuting — CHANGES_REQUESTED** [review]
PR has commented code, what to do with them?

File: ext/resource/loader/loader_test.go
Comment (eddycharly) [reply ext/resource/loader/loader_test.go:1]: Nope, to be fixed

File: ext/resource/convert/convert_test.go
Comment (eddycharly) [reply ext/resource/convert/convert_test.go:1]: Needs fix

**Review by realshuting — APPROVED** [review]

**Pull request merged**